src/tools/tidy/src/rustdoc_html_ids.rs

```diff
@@ -0,0 +1,164 @@
+        if ids.insert(id.to_owned(), 0).is_some() {
+            eprintln!(
+                "=> ID `{}` is defined more than once in the ID map in file `{}`",
+                id, ID_MAP_PATH
+            );
+            *bad = true;
+        }
+    }
+    if ids.is_empty() {
+        eprintln!("=> No IDs were found in rustdoc in file `{}`...", ID_MAP_PATH);
+        *bad = true;
+    }
+    ids
+}
+
+fn check_id(
+    path: &Path,
+    id: &str,
+    ids: &mut HashMap<String, usize>,
+    line_nb: usize,
+    bad: &mut bool,
+) {
+    if id.contains('{') {
+        // This is a formatted ID, no need to check it!
+        return;
+    }
+    let id = id.to_owned();
+    match ids.get_mut(&id) {
+        Some(nb) => *nb += 1,
+        None => {
+            eprintln!(
+                "=> ID `{}` in file `{}` at line {} is missing from `init_id_map`",
+                id,
+                path.display(),
+                line_nb + 1,
+            );
+            *bad = true;
+        }
+    }
+}
+
+fn check_ids(
+    path: &Path,
+    f: &str,
+    ids: &mut HashMap<String, usize>,
+    regex: &Regex,
+    bad: &mut bool,
+    small_section_header_checked: &mut usize,
+) {
+    let mut is_checking_small_section_header = None;
+
+    for (line_nb, line) in f.lines().enumerate() {
+        let trimmed = line.trim_start();
+        // We're not interested in comments or doc comments.
+        if trimmed.starts_with("//") {
+            continue;
+        } else if let Some(start_line) = is_checking_small_section_header {
+            if line_nb == start_line + 2 {
+                check_id(path, trimmed.split('"').skip(1).next().unwrap(), ids, line_nb, bad);
+                is_checking_small_section_header = None;
+            }
+        } else if trimmed.starts_with("write_small_section_header(") {
+            // This function is used to create section: the second argument of the function is an
+            // ID and we need to check it as well, hence this specific check...
+            if trimmed.contains(',') {
+                // This is a call made on one line, so we can simply check it!
+                check_id(path, trimmed.split('"').skip(1).next().unwrap(), ids, line_nb, bad);
+            } else {
+                is_checking_small_section_header = Some(line_nb);
+            }
+            *small_section_header_checked += 1;
+            continue;
+        }
+        for cap in regex.captures_iter(line) {
+            check_id(path, &cap[1], ids, line_nb, bad);
+        }
+    }
+}
+
+pub fn check(path: &Path, bad: &mut bool) {
+    // matches ` id="blabla"`
+    let regex = Regex::new(r#"[\s"]id=\\?["']([^\s\\]+)\\?["'][\s\\>"{]"#).unwrap();
+
+    println!("Checking rustdoc IDs...");
+    let mut ids = extract_ids(&path.join(ID_MAP_PATH), bad);
+    let mut small_section_header_checked = 0;
+    if *bad {
+        return;
+    }
+    super::walk(
+        &path.join("librustdoc/html"),
```

Comment:
From change to search only `src/librustdoc`:

```suggestion
        &path.join("html"),
```